odoo/service/wsgi_server.py

```diff
@@ -108,17 +108,18 @@ def application_unproxied(environ, start_response):
     from werkzeug.middleware.proxy_fix import ProxyFix as ProxyFix_
     # 0.15 also supports port and prefix, but 0.14 only forwarded for, proto
     # and host so replicate that
-    ProxyFix = lambda app: ProxyFix_(app, x_for=1, x_proto=1, x_host=1)
```

Comment:
I have been staring at this with my security hat on and noticed that Odoo sets x_host=1 here, whereas [Werkzeug defaults x_host=0](https://werkzeug.palletsprojects.com/en/0.16.x/middleware/proxy_fix/#werkzeug.middleware.proxy_fix.ProxyFix). It would be a breaking change, but we could address @odony 's concern by forcing users to opt into the dangerous behavior with `--proxy-x-host=1`.